src/features/header/header.tsx

```diff
@@ -0,0 +1,112 @@
+import styled from 'styled-components';
+import { Hamburger } from '#ui/hamburger/hamburger';
+import { Initials } from '#ui/user/user-initials/user-initials';
+import { Username } from '#ui/user/user-name/username';
+import { ReactComponent as SearchIcon } from '../../images/search-original.svg';
+import { ReactComponent as CancelIcon } from '../../images/Icon-Cancel.svg';
+import { useAppDispatch, useAppSelector } from '../../hooks';
+import { toggle } from './header.slice';
+import { menuMockArray } from '../../mock-data/mock-data-menu';
+
+export const Header = () => {
+  const dispatch = useAppDispatch();
+  const { hamburgerOpen } = useAppSelector((state) => state.hamburgerMenu);
+
+  return (
+    <HeaderWrapper>
+      <Hamburger
+        items={menuMockArray.map((item) => item)}
+        onClick={() => dispatch(toggle())}
+      />
+
+      <SearchArea>
+        <SearchInput
+          type="text"
+          placeholder="Search..."
+          maxLength={100}
+          style={{ visibility: !hamburgerOpen ? 'visible' : 'hidden' }}
+        />
+        <ButtonsDiv>
+          <CloseButton
+            type="button"
+            style={{ visibility: !hamburgerOpen ? 'visible' : 'hidden' }}
+          >
+            <CancelIcon />
+          </CloseButton>
+          <SearchButton type="button">
+            <SearchIcon />
+          </SearchButton>
+        </ButtonsDiv>
+      </SearchArea>
+      <UserWrapper>
+        <Initials firstName="Maryia" secondName="dauhan" />
+        <Username firstName="Maryia" secondName="dauhan" />
+      </UserWrapper>
+    </HeaderWrapper>
+  );
+};
+
+const HeaderWrapper = styled.div`
+  all: unset;
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: center;
+  background-color: var(--system-primary-color);
+  height: 84px;
+`;
+
+const SearchArea = styled.div`
+  all: unset;
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  justify-content: space-between;
+  width: -webkit-fill-available;
+`;
+
+const SearchInput = styled.input`
+  all: unset;
+  height: 84px;
+  width: -webkit-fill-available;
```

Comment:
`-webkit-fill-available` usually used on Safari iOS to make height 100% of available space

for other cases better _not_ use it because it's not standardized

possible alternatives: 100%, 100vw, flex-grow: 1